rotkehlchen/api/rest.py

```diff
@@ -5346,3 +5346,66 @@ def get_historical_prices_per_asset(
             'no_prices_timestamps': no_prices_ts,
             'rate_limited_prices_timestamps': rate_limited_prices_ts,
         })
+
+    @async_api_call()
+    def force_refetch_evm_transactions(
+            self,
+            to_timestamp: Timestamp,
+            from_timestamp: Timestamp,
+            address: ChecksumEvmAddress | None = None,
+            evm_chain: EVM_CHAIN_IDS_WITH_TRANSACTIONS_TYPE | None = None,
+    ) -> dict[str, Any]:
+        """Re-query EVM transactions for a given time range, adding only missing transactions.
+
+        This function requests a force refetch of transactions for the specified time range,
+        bypassing the normal query range checks. This can be useful in cases where transactions
+        might have been missed due to API issues or other temporary problems.
+        """
+        log.debug(
+            'Force refetching EVM transactions',
+            from_ts=from_timestamp,
+            to_ts=to_timestamp,
+            chain=evm_chain.name if evm_chain else 'all supported chains',
+            address=address or 'all addresses',
+        )
+
+        transaction_count = 0
+        db_evmtx = DBEvmTx(self.rotkehlchen.data.db)
+        chains_to_query: list[EVM_CHAIN_IDS_WITH_TRANSACTIONS_TYPE] = [evm_chain] if evm_chain else list(EVM_CHAIN_IDS_WITH_TRANSACTIONS)  # noqa: E501
+        for chain_id in chains_to_query:
+            chain_manager = self.rotkehlchen.chains_aggregator.get_evm_manager(chain_id)
+            if address:
+                addresses_to_query: tuple[ChecksumEvmAddress, ...] = (address,)
+            else:
+                with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+                    addresses_to_query = self.rotkehlchen.data.db.get_blockchain_accounts(cursor).get(chain_manager.node_inquirer.blockchain)  # noqa: E501
+
+            if addresses_to_query is None or len(addresses_to_query) == 0:
```

Comment:
When can `addresses_to_query` be None here? Isn't it always a tuple of addresses and at worst case an empty tuple? Or am I missing something?

```suggestion
            if addresses_to_query is None or len(addresses_to_query) == 0:
```

Reply:
my bad. i didn't get the return type of `.get()` in this case. assumed it'll return None.